elastalert/elastalert.py

```diff
@@ -775,13 +776,20 @@ def init_rule(self, new_rule, new=True):
                 continue
             new_rule[prop] = rule[prop]
 
-        if self.is_five():
-            self.modify_rule_for_ES5(new_rule)
+        self.modify_rule_for_ES5(new_rule)
 
         return new_rule
 
     @staticmethod
     def modify_rule_for_ES5(new_rule):
+        # Get ES version per rule
+        rule_es = elasticsearch_client(new_rule)
+        if rule_es.info()['version']['number'].startswith('5'):
+            new_rule['five'] = True
+        else:
+            new_rule['five'] = False
+            return
```

Comment:
Except we now call modify_rule_for_es5 on every rule, and we don't want to pop the top level 'query' for non-five.